app/src/main/java/com/ybsystem/tweethub/application/TweetHubApp.java

```diff
@@ -0,0 +1,89 @@
+package com.ybsystem.tweethub.application;
+
+import android.annotation.SuppressLint;
+import android.app.Application;
+
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.ybsystem.tweethub.models.entities.Account;
+import com.ybsystem.tweethub.models.entities.AppData;
+import com.ybsystem.tweethub.models.entities.Via;
+import com.ybsystem.tweethub.models.entities.twitter.TwitterUser;
+import com.ybsystem.tweethub.storages.PrefEntity;
+
+import twitter4j.Twitter;
+import twitter4j.TwitterFactory;
+import twitter4j.auth.AccessToken;
+
+/**
+ * Application class of TweetHub
+ * Defines global variables used in the application
+ */
+public class TweetHubApp extends Application {
+
+    @SuppressLint("StaticFieldLeak")
+    private static TweetHubApp sTweetHubApp;
+
+    private AppCompatActivity mActivity;
+    private AppData mAppData;
+    private Twitter mTwitter;
+
+    @Override
+    public void onCreate() {
+        super.onCreate();
+
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Make the enclosing method "static" or remove this set. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://192.168.11.19:9000/coding_rules#rule_key=squid%3AS2696)